src/main/java/core/basesyntax/impl/StorageImpl.java

```
@@ -3,17 +3,53 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int currentSize;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[SIZE];
+        this.values = (V[]) new Object[SIZE];
+        this.currentSize = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = findKeyIndex(key);
+        if (index != -1) {
+            values[index] = value;
+            return;
+        }
+
+        if (currentSize < SIZE) {
+            keys[currentSize] = key;
+            values[currentSize] = value;
+            currentSize++;
+        }
```

Behavior note / suggestion: the code respects the capacity constant (`SIZE`) but silently ignores `put` calls when the storage is full (the `if (currentSize < SIZE)` branch simply skips insertion when full). The requirements ask to enforce capacity; while not required to throw, it should be documented or handled explicitly.

Suggestion: either document that additional puts are ignored when full, or throw a checked/unchecked exception (for example `IllegalStateException`) when attempting to put beyond capacity. If you change to an `Entry[]` design, ensure you preserve the capacity logic (and keep `currentSize` updated only on new insertions).